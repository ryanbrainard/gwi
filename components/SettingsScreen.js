import React from "react";
import { Switch, Text, View } from "react-native";
import Settings from "../models/Settings";

export default class SettingsScreen extends React.Component {
  static navigationOptions = {
    title: "Settings"
  };

  constructor(props) {
    super(props);
    this.state = {
      settings: {}
    };
  }

  componentWillMount() {
    Settings.getAll().then(settings => {
      this.setState({ settings: settings });
    });
  }

  toggleFor(key) {
    return value => {
      Settings.set(key, value).then(() =>
        this.setState({
          settings: Object.assign(this.state.settings, {
            [key]: value
          })
        })
      );
    };
  }

  render() {
    return (
      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
        {Object.entries(this.state.settings).map(([key, value]) => (
          <View key={key}>
            <Text>{key}</Text>
            <Switch
              onValueChange={this.toggleFor(key).bind(this)}
              value={value}
            />
          </View>
        ))}
      </View>
    );
  }
}
