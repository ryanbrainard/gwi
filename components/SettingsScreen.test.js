import React from "react";
import SettingsScreen from "./SettingsScreen";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<SettingsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
