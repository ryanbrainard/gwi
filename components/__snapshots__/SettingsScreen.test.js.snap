// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
